agent/xds/clusters.go

```diff
@@ -443,19 +443,23 @@ func (s *ResourceGenerator) makeDestinationClusters(cfgSnap *proxycfg.ConfigSnap
 		svcConfig, _ := serviceConfigs[svcName]
 		dest := svcConfig.Destination
 
-		// If IP, create a cluster with the fake name.
-		if dest.HasIP() {
-			opts := clusterOpts{
-				name:            connect.ServiceSNI(svcName.Name, "", svcName.NamespaceOrDefault(), svcName.PartitionOrDefault(), cfgSnap.Datacenter, cfgSnap.Roots.TrustDomain),
-				addressEndpoint: dest,
+		for _, address := range dest.Addresses {
+			// If IP, create a cluster with the fake name.
+			if structs.IsIP(address) {
```

Comment:
For this PR I kept the dynamic cluster. I was going to remove/update that in a follow-up PR because I think there is some nuance in the cluster configuration needed. I think hostnames will always require something different, i.e. LOGICAL_DNS vs STATIC clusters.

Reply:
👍 for another PR